src/checkers/inference/model/serialization/ToStringSerializer.java

```diff
@@ -24,49 +28,65 @@
 import checkers.inference.model.VariableSlot;
 
 /**
- * This Serializer is meant only to convert constraints and variables to
- * human readable strings.  It is used currently by the DebugSolver.
+ * This Serializer is meant only to convert constraints and variables to human readable strings. It
+ * is used currently by the DebugSolver.
  */
 public class ToStringSerializer implements Serializer<String, String> {
     private final boolean showAstPaths;
     private boolean showVerboseVars;
 
-    private int indent = 0;
-    public static final String INDENT_STRING = "    ";
+    private int indentationLevel = 0;
+
+    // 2 spaces per indentation
+    public static final String INDENT = "  ";
+
+    // stores N concatenations of INDENT at index N, where index 0 stores empty string
+    // the index is the indentation level
+    protected final List<String> indentStrings = new ArrayList<>();
```

Comment:
Why protected?

Reply:
Making it private. Are code comments insufficient as documentation? It already explains what the list is for...

Reply:
Yeah, not that clear... the comment now explains what the list is, but how it's going to be used is still not clear. Without looking at the source code, I don't understand why having indentation level, INDENT, and indentStrings. All these seem overwhelming to me.

Reply:
is the most up to date comments more clear?